rust/lancedb/resources/remote-catalog-open-api.yaml

```diff
@@ -0,0 +1,284 @@
+openapi: 3.1.1
+info:
+  title: LanceDB REST Catalog API
+  version: 0.0.1
+  description: REST interface for managing LanceDB databases and tables
+
+paths:
+  /v1/databases:
+    get:
+      tags: [Database]
+      summary: List databases
+      parameters:
+        - $ref: '#/components/parameters/startAfter'
+        - $ref: '#/components/parameters/limit'
+      responses:
+        200:
+          description: Database list response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DatabaseList'
+
+    post:
+      tags: [Database]
+      summary: Create new database
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDatabaseRequest'
+      responses:
+        201:
+          description: Database created
+          headers:
+            Location:
+              description: URI of created database
+              schema: {type: string}
+
+    delete:
+      tags: [ Database ]
+      summary: Delete all databases
+      responses:
+        204:
+          description: All databases deleted
+
+  /v1/databases/{dbName}:
+    get:
+      tags: [Database]
+      summary: Get database metadata
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+      responses:
+        200:
+          description: Database metadata
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DatabaseMetadata'
+
+    delete:
+      tags: [Database]
+      summary: Delete database
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+      responses:
+        204:
+          description: Database deleted
+
+    put:
```

Comment:
regarding the verb here, I remember IRC uses POST but Gravitino uses PUT to express an update. Personally I am always in favor of POST because every system supports POST, GET & DELETE, and other verbs like PUT, PATCH gets strange behavior definitions in some systems. But I am not too strongly opinionated on this.